src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+Existing Yubikey devices have already been enabled in RHEL and Fedora but the
+approach is completely different. The existing solution uses
+`OATH-TOTP <https://en.wikipedia.org/wiki/Time-based_one-time_password>`__ and
+`OATH-HOTP <https://en.wikipedia.org/wiki/HMAC-based_one-time_password>`__ to
+generate a one-time password (OTP) that is sent to the server as a second
+factor authentication. In other words, the authentication checks happen on the
+server. Moreover, this solution relies on a plugin for the 389 directory
+server, so it only works with IPA for IPA users.
+
+The approach explained in this document relies on the 
+`FIDO2 <https://en.wikipedia.org/wiki/FIDO2_Project>`__ standard, which uses a
+public-private key scheme to perform the authentication checks on the client.
+Besides, the solution is general enough to accept different LDAP servers like
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
+
+Implementation details
+----------------------
+
+The implementation will loosely follow existing smartcard authentication in
+SSSD. On top of that, and as mentioned before, the information will be stored
+in two places: in the LDAP server and on the passkey. The server will contain
+the public user credentials, including the user key mapping data, while the
+passkey device will store the private key necessary to authenticate the user.
+
+Key registration
+****************
+The first step is to register the user with a key. SSSD will provide a helper
+utility to handle the passkeys, and it can be used to create the key mapping
+string (similar to pamu2fcfg). Instead of calling the helper utility from the
+command line the system administrator would use 
+`sssctl <https://docs.pagure.org/sssd.sssd/design_pages/sssctl.html>`__. This
+would enable to unify the user experience. Then, the output of this command can
+be set to the LDAP server in a dedicated attribute. For AD we could use the
+`altSecurityIdentities <https://learn.microsoft.com/en-us/windows/win32/adschema/a-altsecurityidentities>`__
+with a prefix (i.e. passkey). For FreeIPA we would use a new attribute called
+ipapasskey from the ipapasskeyuser objectclass. For any other LDAP server we
+would use the passkey attribute.
+
+The format for
+`server-side credentials <https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html>`__
+for the key mapping will be
+``passkey:credentialId,pemPublicKey``.
+
+The format for
+`discoverable credentials <https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html>`__ for the key mapping will be
+``passkey:credentialId,pemPublicKey,userId``.
+
+IPA process
++++++++++++
+FreeIPA would provide a direct way to register the passkey attributes to the
```

Comment:
I think for IPA you should provide a link to readthedocs page. Since IPA part is not merged yet (chicken and egg problem with SSSD release with passkey yet), we should probably update this part of the design once it is published. It is definitely good to replace 'would provide' and below 'can be done' by more concrete statement based on the IPA design.  Please see IPA design here: https://github.com/flo-renaud/freeipa/blob/passkey/doc/designs/passkeys.md

Reply:
I'm keeping this open until until IPA part is merged.

Reply:
Opened https://github.com/SSSD/sssd.io/pull/69 to include the link